### PR TITLE
[11.0] [fix] purchase_discount

### PR DESCRIPTION
**Pull request by hugosantosred**
Use all the decimal positions from _get_discounted_price_unit to compute
the purchase line amount, for not lose decimal precision.
This can lead to unbalanced amounts between the purchase and the related invoice

https://github.com/OCA/purchase-workflow/issues/558

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@legalsylvain what do you think about this?

File: purchase_discount/tests/test_purchase_discount.py [hunk 1]
```diff
@@ -126,3 +141,17 @@ def test_invoice(self):
...
+            'purchase_id': self.purchase_order_check_amount.id,
+        })
+        invoice.purchase_order_change()
+        line = invoice.invoice_line_ids.filtered(
+            lambda x: x.purchase_line_id == self.po_line_check_amount
+        )
+        line.quantity = self.po_line_check_amount.product_qty
+        self.assertEqual(line.discount, 1)
+        self.assertAlmostEqual(line.price_subtotal,
+                               self.po_line_check_amount.price_subtotal)
```
Comment (pedrobaeza) [inline purchase_discount/tests/test_purchase_discount.py:157]: You must use fixed data in tests (for example, `15.03`), not references to other fields.

**Comment by legalsylvain** [comment]
Hi.

About the original bug, mentionned here #558. 

> Price_unit: 19.75 * discount: 0.5 = 9,875 Rounding (????) = 9,88 * Units = 30 = 296.40
The correct calculation:
Price_unit: 19.75 * discount: 0.5 = 9,875 * Units:30 = 296.25 Rounding here!!!!

I don't see as a bug, and for me, should be an option on the supplier.

I'm using a lot discount (1 & 2) on purchase module. In fact, some of my suppliers are using the first method of rounding, other are using the second one.

Indeed the default method should be the second one, as it is a more accurate.

My proposal : include the code of purchase_rounding_method in the module purchase_triple_discount.

https://github.com/OCA/purchase-workflow/tree/8.0/purchase_supplier_rounding_method

What do you think ?

**Comment by vincent-hatakeyama** [comment]
The sale module has a discount on sales order lines, and the price on the line would be 296.25 in the example provided. The purchase module discount is expected to work the same way, but without this fix it does not.

**Comment by vincent-hatakeyama** [comment]
I still think this should be merged to fix issue #850. It seems that some work need to be done to include it thought.

**Pull request closed without merging**